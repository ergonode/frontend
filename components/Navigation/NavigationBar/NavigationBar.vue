/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <nav class="navigation-bar">
        <section class="navigation-bar__section">
            <slot name="leftSection" />
        </section>
        <section class="navigation-bar__section">
            <slot name="rightSection" />
        </section>
    </nav>
</template>

<script>
export default {
    name: 'NavigationBar',
};
</script>

<style lang="scss" scoped>
    @import "~/assets/scss/navigation/navigationbar.scss";
</style>
