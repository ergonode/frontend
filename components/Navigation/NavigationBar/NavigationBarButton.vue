/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <button :class="['bar-button', {'bar-button--selected': isSelected}]">
        <slot />
    </button>
</template>

<script>
export default {
    name: 'NavigationBarButton',
    props: {
        isSelected: {
            type: Boolean,
            default: false,
        },
    },
};
</script>

<style lang="scss" scoped>
    .bar-button {
        position: relative;
        display: flex;
        justify-content: center;
        align-items: center;
        border: none;
        padding: 0 24px;
        background-color: transparent;
        cursor: pointer;
        outline: none;

        &:hover, &--selected {
            background-color: rgba($GRAPHITE, 0.5);
        }

        .title {
            color: $WHITE;
            font: $FONT_SEMI_BOLD_14_20;
            margin-left: 8px;
        }
    }
</style>
