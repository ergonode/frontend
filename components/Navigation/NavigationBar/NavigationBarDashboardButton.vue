/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarButton @click.native="onClick">
        <img
            class="part__element"
            src="~assets/images/logo.svg"
            alt="ergonode"
            title="ergonode">
    </NavigationBarButton>
</template>

<script>
export default {
    name: 'NavigationBarDashboardButton',
    components: {
        NavigationBarButton: () => import('~/components/Navigation/NavigationBar/NavigationBarButton'),
    },
    methods: {
        onClick() {
            this.$router.push('/dashboard');
        },
    },
};
</script>
