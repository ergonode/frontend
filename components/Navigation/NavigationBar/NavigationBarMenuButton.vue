/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarButton @click.native="onClick">
        <Component
            :is="menuIconComponent"
            fill-color="#fff"
            :state="leftDoubleArrow" />
    </NavigationBarButton>
</template>

<script>
import { Arrow } from '~/model/icons/Arrow';
import IconHamburgerList from '~/components/Icon/Menu/IconHamburgerList';
import IconArrowDouble from '~/components/Icon/Arrows/IconArrowDouble';
import IconHamburger from '~/components/Icon/Menu/IconHamburger';

export default {
    name: 'NavigationBarMenuButton',
    components: {
        NavigationBarButton: () => import('~/components/Navigation/NavigationBar/NavigationBarButton'),
    },
    props: {
        value: {
            type: Number,
            required: true,
        },
    },
    data() {
        return {
            leftDoubleArrow: Arrow.LEFT,
        };
    },
    computed: {
        menuIconComponent() {
            switch (this.value) {
            case 2:
                return IconHamburgerList;
            case 1:
                return IconArrowDouble;
            default:
                return IconHamburger;
            }
        },
    },
    methods: {
        onClick() {
            const newState = this.value > 0 ? this.value - 1 : 2;

            this.$emit('state', newState);
        },
    },
};
</script>
