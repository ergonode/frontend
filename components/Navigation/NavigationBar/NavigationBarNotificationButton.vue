/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarButton>
        <IconBell :fill-color="whiteColor" />
    </NavigationBarButton>
</template>

<script>
import { WHITE } from '~/assets/scss/_variables/_colors.scss';

export default {
    name: 'NavigationBarNotificationButton',
    components: {
        NavigationBarButton: () => import('~/components/Navigation/NavigationBar/NavigationBarButton'),
        IconBell: () => import('~/components/Icon/Menu/IconBell'),
    },
    computed: {
        whiteColor() {
            return WHITE;
        },
    },
};
</script>
