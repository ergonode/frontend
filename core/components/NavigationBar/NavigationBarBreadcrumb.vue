/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="navigation-bar-breadcrumb">
        <Component
            :is="breadcrumbIcon"
            v-if="breadcrumb.icon"
            :fill-color="breadcrumbIconFillColor" />
        <NuxtLink
            v-if="breadcrumb.routePath"
            class="navigation-bar-breadcrumb__link"
            :to="breadcrumb.routePath"
            v-text="breadcrumb.title" />
        <span
            v-else
            class="navigation-bar-breadcrumb__title"
            v-text="breadcrumb.title" />
        <span class="navigation-bar-breadcrumb__divider">/</span>
    </div>
</template>

<script>
import { GREEN } from '~/assets/scss/_variables/_colors.scss';

export default {
    name: 'NavigationBarBreadcrumb',
    props: {
        breadcrumb: {
            type: Object,
            required: true,
        },
    },
    computed: {
        breadcrumbIconFillColor() {
            return GREEN;
        },
        breadcrumbIcon() {
            return () => import(`~/components/Icon/Menu/Icon${this.breadcrumb.icon}`);
        },
    },
};
</script>

<style lang="scss" scoped>
    .navigation-bar-breadcrumb {
        display: grid;
        grid-auto-flow: column;
        grid-column-gap: 8px;
        align-items: center;
        font: $FONT_MEDIUM_14_20;

        &__title, &__link {
            color: $WHITE;
        }

        &__link {
            text-decoration: none;
        }

        &__divider {
            color: $GREY_DARK;
        }
    }
</style>
