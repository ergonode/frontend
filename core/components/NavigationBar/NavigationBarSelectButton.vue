/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NavigationBarButton
        :is-selected="isFocused"
        @click.native="onClick">
        <slot name="input" />
        <FadeTransition>
            <div
                v-if="isFocused"
                ref="menu"
                class="select-content">
                <slot name="selectContent" />
            </div>
        </FadeTransition>
    </NavigationBarButton>
</template>

<script>
import {
    isMouseOutOfBoundsElement,
    getPositionForBrowser,
} from '~/model/drag_and_drop/helpers';

export default {
    name: 'NavigationBarSelectButton',
    components: {
        NavigationBarButton: () => import('~/core/components/NavigationBar/NavigationBarButton'),
        FadeTransition: () => import('~/core/components/Transitions/FadeTransition'),
    },
    data() {
        return {
            isFocused: false,
            isClickedOutside: false,
        };
    },
    destroyed() {
        window.removeEventListener('click', this.onClickOutside);
    },
    methods: {
        onClick() {
            if (!this.isFocused) {
                window.addEventListener('click', this.onClickOutside);
            }
        },
        onClickOutside(event) {
            let isClickedInsideMenu = false;

            if (this.isFocused) {
                const { xPos, yPos } = getPositionForBrowser(event);
                const { menu } = this.$refs;

                isClickedInsideMenu = !isMouseOutOfBoundsElement(menu, xPos, yPos);
                if (!isClickedInsideMenu) {
                    this.isFocused = false;
                }
            } else {
                this.isFocused = !this.isFocused;
            }

            if (!this.isFocused) {
                window.removeEventListener('click', this.onClickOutside);
            }

            this.$emit('focus', { focus: this.isFocused, isClickedInsideMenu });
        },
    },
};
</script>

<style lang="scss" scoped>
    .select-content {
        position: absolute;
        top: 48px;
        right: 0;
        z-index: $Z_INDEX_NAV;
        display: flex;
        background-color: $WHITE;
        box-shadow: $ELEVATOR_2_DP;
        transition: all 0.4s ease;
        color: $GRAPHITE;
        cursor: default;
    }
</style>
