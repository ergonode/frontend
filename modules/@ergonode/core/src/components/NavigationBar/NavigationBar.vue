/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <nav class="navigation-bar">
        <section class="navigation-bar__breadcrumbs">
            <slot name="breadcrumbs" />
        </section>
        <section class="navigation-bar__actions">
            <slot name="actions" />
        </section>
    </nav>
</template>

<script>
export default {
    name: 'NavigationBar',
};
</script>

<style lang="scss" scoped>
    .navigation-bar {
        display: flex;
        justify-content: space-between;
        height: 48px;
        background-color: $GRAPHITE_COAL;

        &__breadcrumbs {
            display: grid;
            grid-auto-flow: column;
            grid-column-gap: 8px;
            align-items: center;
            padding: 0 16px;
            box-sizing: border-box;
        }

        &__actions {
            display: grid;
            grid-auto-flow: column;
            grid-column-gap: 1px;
            background-color: $GRAPHITE_DARK;
            border-left: 1px solid $GRAPHITE_DARK;
        }
    }
</style>
