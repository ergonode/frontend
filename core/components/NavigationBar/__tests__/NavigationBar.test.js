/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import { shallowMount, createLocalVue } from '@vue/test-utils';
import NavigationBar from '~/core/components/NavigationBar/NavigationBar';

const localVue = createLocalVue();
describe('NavigationBar', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(NavigationBar, {
            localVue,
        });
    });

    it('Component is named well', () => {
        expect(typeof NavigationBar.name).toBe('string');
        expect(NavigationBar.name).toEqual('NavigationBar');
    });

    it('Has no items', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
