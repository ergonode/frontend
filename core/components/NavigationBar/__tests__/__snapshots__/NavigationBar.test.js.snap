// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Has no items 1`] = `
<nav class="navigation-bar">
  <section class="navigation-bar__section"></section>
  <section class="navigation-bar__section"></section>
</nav>
`;
